src/frontend/apps/lti_site/components/VideoWidgetProvider/widgets/DownloadVideo/index.tsx

```diff
@@ -78,13 +103,59 @@ export const DownloadVideo = () => {
     value: string;
   }>(options[options.length - 1]);
 
+  const [toggleAllowDownload, setToggleAllowDownload] = useState(
+    video.show_download,
+  );
+
+  const [disabledToggle, setDisabledToggle] = useState(false);
+
+  const videoMutation = useUpdateVideo(video.id, {
+    onSuccess: (videoUpdated: Video) => {
+      setToggleAllowDownload(videoUpdated.show_download);
+      toast.success(
+        intl.formatMessage(
+          videoUpdated.show_download
+            ? messages.allowDownloadToggleSuccess
+            : messages.disallowDownloadToggleSuccess,
+        ),
+        {
+          position: 'bottom-center',
+        },
+      );
+    },
+    onError: (err) => {
+      report(err);
+      toast.error(intl.formatMessage(messages.allowDownloadToggleFail), {
+        position: 'bottom-center',
+      });
+    },
```

Comment:
You should probably reset the switch value also

Reply:
@RVassili as implemented there is no need to do that.